internal/configs/mock_provider.go

```diff
@@ -172,6 +206,13 @@ type Override struct {
 	Target *addrs.Target
 	Values cty.Value
 
+	// By default, overridden computed values are ignored during planning,
+	// and the computed values are set to unknown to simulate the behavior
+	// of a real plan. This attribute indicates that the computed values
+	// should be overridden with the values specified in the override block,
+	// even when planning.
+	useForPlan *bool
```

Comment:
sorry, I didn't notice this before - I don't think this needs to be a pointer. You can just make it a normal bool, and public so callers can read it directly.

We don't need to distinguish between "not set" and explicitly set to false right?